File: Others/3- Euclidean algorithms for GCD/Euclidean.py
def ExtendedGCD(a, b):
    if a == 0:
        return b, 0, 1
    gcd, x1, y1 = ExtendedGCD(b % a, a)
    x = y1 - (b // a) * x1
    y = x1
    return gcd, x, y


def BasicGCD(a, b):
    if a == 0:
        return b
    return BasicGCD(b % a, a)

File: Others/3- Euclidean algorithms for GCD/Euclidean.java
public class Euclidean {

    public static int BasicGCD(int a, int b) {
        if (a == 0)
            return b;
        return BasicGCD(b%a, a);
    }

    public static int ExtendedGCD(int a, int b, int x, int y) {
        if (a == 0) {
            x = 0;
            y = 1;
            return b;
        }
        int x1=1, y1=1;
        int gcd = ExtendedGCD(b%a, a, x1, y1);
        x = y1 - (b/a) * x1;
        y = x1;
        return gcd;
    }
}
